### PR TITLE
[MIG] base_multi_image: Migration to 12.0

### PR DESCRIPTION
**Pull request by shepilov-vladislav**
I squash all commits to one from here https://github.com/OCA/server-tools/pull/1528

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Runbot is still red. Check if the warnings I put in the other PR are still there.

**Comment by shepilov-vladislav** [comment]
> Runbot is still red. Check if the warnings I put in the other PR are still there.

Ok, will check.

**Comment by shepilov-vladislav** [comment]
> Runbot is still red. Check if the warnings I put in the other PR are still there.

@pedrobaeza I checked these warnings - it's came from `base_multi_image` module. But the module `base_multi_image` is located in a different repository(https://github.com/OCA/server-tools).

**Comment by shepilov-vladislav** [comment]
> Runbot is still red. Check if the warnings I put in the other PR are still there.

@pedrobaeza Build is green now! :)

**Comment by pedrobaeza** [comment]
Yeah, great, but you must preserve commit history following technical method explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0

**Comment by pedrobaeza** [comment]
The only thing to squash are migration commits, not whole history.

**Comment by shepilov-vladislav** [comment]
> The only thing to squash are migration commits, not whole history.

Sorry, @pedrobaeza , I'm not sure that I understand what needs to be done. When migrating, I followed the instructions described [here](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0)

**Comment by pedrobaeza** [comment]
Are you migrating from 11.0 or another branch? If another, you must change 11.0 by the source branch name, or you will end up with no commit history imported, like it's happening in this PR.

**Comment by shepilov-vladislav** [comment]
> Are you migrating from 11.0 or another branch? If another, you must change 11.0 by the source branch name, or you will end up with no commit history imported, like it's happening in this PR.

I create my branch from `12.0` branch, move module from https://github.com/OCA/server-tools/pull/1528 add some fixes for README and runbot warnings and commit all files with one commit. What I'm did wrong?

**Comment by ivantodorovich** [comment]
Hi @shepilov-vladislav 

The goal when migrating a module to a new version/branch is to keep the original commit history + one migration commit with the necessary changes.

To achieve that, instead of moving the module files, you need to use the commands described here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0

Specifically:
```
$ git clone https://github.com/OCA/$REPO -b 12.0
$ git checkout -b 12.0-mig-$MODULE origin/12.0
$ git format-patch --keep-subject --stdout origin/12.0..origin/11.0 -- $MODULE | git am -3 --keep
```

For instance, look at the commits of this merged migration PR:
https://github.com/OCA/server-tools/pull/1560/commits
image

The original commit history is still there, + 1 extra migration commit

**Comment by shepilov-vladislav** [comment]
@ivantodorovich Great TY for explanation!

**Comment by shepilov-vladislav** [comment]
> Are you migrating from 11.0 or another branch? If another, you must change 11.0 by the source branch name, or you will end up with no commit history imported, like it's happening in this PR.

@pedrobaeza ok, I fixed commit history.

**Comment by shepilov-vladislav** [comment]
> Hey @shepilov-vladislav, thank you for your Pull Request.
> 
> It looks like some users haven't signed our **C**ontributor **L**icense **A**greement, yet.
> You can read and sign our full Contributor License Agreement here: http://odoo-community.org/page/cla
> Here is a list of the users:
> 
> * Vladislav Shepilov [vladislavshepilov@vladislavs-macbook-pro.local](mailto:vladislavshepilov@vladislavs-macbook-pro.local) (no github login found)
> 
> Appreciation of efforts,
> OCA CLAbot

Fixed.

**Comment by shepilov-vladislav** [comment]
So, can we merge this PR? Or need to do anything else?

**Comment by pedrobaeza** [comment]
It should be reviewed by qualified reviewers (I only did an "administrative" review). You can encourage the review if you review other PRs and ask in exchange that they review yours.

File: base_multi_image/views/image_view.xml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                           type="delete"
+                           style="position: absolute; right: 0; padding: 4px; diplay: inline-block">X</a>
+                        <div class="oe_module_vignette">
+                            <a type="open">
+                              <img
+                                  t-att-alt="record.name"
+                                  t-att-src="kanban_image(
+                                      'base_multi_image.image',
+                                      'image_small',
+                                      record.id.value)"
```
Comment (FFernandez-PlanetaTIC) [inline base_multi_image/views/image_view.xml:111]: In locales with thousand separator different from comma, like dot in Spain, image URL is invalid:

image

```
Traceback (most recent call last):
  File "eggs/Werkzeug-0.11.15-py3.5.egg/werkzeug/serving.py", line 205, in run_wsgi
    execute(self.server.app)
  File "eggs/Werkzeug-0.11.15-py3.5.egg/werkzeug/serving.py", line 193, in execute
    application_iter = app(environ, start_response)
  File "parts/OCA/OCB/odoo/service/wsgi_server.py", line 142, in application
    return application_unproxied(environ, start_response)
  File "parts/OCA/OCB/odoo/service/wsgi_server.py", line 117, in application_unproxied
    result = odoo.http.root(environ, start_response)
  File "parts/OCA/OCB/odoo/http.py", line 1320, in __call__
    return self.dispatch(environ, start_response)
  File "parts/OCA/OCB/odoo/http.py", line 1293, in __call__
    return self.app(environ, start_wrapped)
  File "eggs/Werkzeug-0.11.15-py3.5.egg/werkzeug/wsgi.py", line 599, in __call__
    return self.app(environ, start_response)
  File "parts/OCA/OCB/odoo/http.py", line 1488, in dispatch
    result = ir_http._dispatch()
  File "parts/OCA/OCB/addons/auth_signup/models/ir_http.py", line 19, in _dispatch
    return super(Http, cls)._dispatch()
  File "parts/OCA/OCB/addons/web_editor/models/ir_http.py", line 22, in _dispatch
    return super(IrHttp, cls)._dispatch()
  File "parts/OCA/OCB/addons/http_routing/models/ir_http.py", line 403, in _dispatch
    result = super(IrHttp, cls)._dispatch()
  File "parts/OCA/OCB/addons/utm/models/ir_http.py", line 26, in _dispatch
    response = super(IrHttp, cls)._dispatch()
  File "parts/OCA/OCB/odoo/addons/base/models/ir_http.py", line 207, in _dispatch
    return cls._handle_exception(e)
  File "parts/OCA/OCB/addons/utm/models/ir_http.py", line 31, in _handle_exception
    response = super(IrHttp, cls)._handle_exception(exc)
  File "parts/OCA/OCB/odoo/addons/base/models/ir_http.py", line 177, in _handle_exception
    return request._handle_exception(exception)
  File "parts/OCA/OCB/odoo/http.py", line 776, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "parts/OCA/OCB/odoo/http.py", line 314, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "parts/OCA/OCB/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "parts/OCA/OCB/odoo/addons/base/models/ir_http.py", line 203, in _dispatch
    result = request.dispatch()
  File "parts/OCA/OCB/odoo/http.py", line 835, in dispatch
    r = self._call_function(**self.params)
  File "parts/OCA/OCB/odoo/http.py", line 346, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "parts/OCA/OCB/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "parts/OCA/OCB/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "parts/OCA/OCB/odoo/http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "parts/OCA/OCB/odoo/http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "parts/OCA/OCB/addons/web/controllers/main.py", line 1076, in content_image
    default_mimetype='image/png', related_id=related_id, access_mode=access_mode, access_token=access_token)
  File "parts/OCA/OCB/addons/web/controllers/main.py", line 435, in binary_content
    env=env)
  File "parts/OCA/OCB/odoo/addons/base/models/ir_http.py", line 298, in binary_content
    obj = env[model].browse(int(id))
ValueError: invalid literal for int() with base 10: '1.616' - - -
```

This line should be _record.id.raw_value_

**Review by FFernandez-PlanetaTIC — CHANGES_REQUESTED** [review]
Hi @shepilov-vladislav ,

Except for the issue commented, it looks good to me.

Thanks!

**Comment by jesusVMayor** [comment]
LGTM, @shepilov-vladislav can you make the change suggested by @FFernandez-PlanetaTIC  ?

**Comment by shepilov-vladislav** [comment]
So old

**Pull request closed without merging**